packages/@lwc/engine-core/src/framework/reporting.ts

```diff
@@ -49,6 +50,10 @@ export interface StylesheetMutationPayload extends BasePayload {
 // eslint-disable-next-line @typescript-eslint/no-empty-object-type
 export interface ConnectedCallbackWhileDisconnectedPayload extends BasePayload {}
 
+export interface RenderModeMismatchPayload extends BasePayload {
+    mode: RenderMode;
```

Comment:
See internal usage [here](https://gitcore.soma.salesforce.com/core-2206/core-public/blob/7792cd9d5825958d78d098479a69a142437b3eae/core/ui-force-components/modules/force/lwcReportingService/lwcReportingService.js#L58). This can just be an integer so it doesn't batter whether it's a `ShadowMode` or `RenderMode`.

Reply:
Should we call this `componentMode` or something, so that we don't have to remember whether it's the component or the template using this mode? Although I guess we don't really care about which way around they are; the mismatch itself is the more important detail.